modules/plugins/languages/typst.nix

```diff
@@ -167,6 +167,48 @@ in {
           };
         };
       };
+      typst-concealer = {
+        enable =
+          mkEnableOption ''
+            [typst-concealer]: https://github.com/PartyWumpus/typst-concealer
+
+            Inline typst preview for Neovim via [typst-concealer]
+          ''
+          // {default = false;};
+
+        setupOpts = mkPluginSetupOption "typst-concealer" {
+          do_diagnostics = mkOption {
+            type = nullOr bool;
+            description = ''Should typst-concealer provide diagnostics on error?'';
+            default = !cfg.lsp.enable;
+          };
+          color = mkOption {
+            type = nullOr str;
+            description = ''What color should typst-concealer render text/stroke with? (only applies when styling_type is "colorscheme")'';
+            default = null;
+          };
+          enabled_by_default = mkOption {
+            type = nullOr bool;
+            description = ''Should typst-concealer conceal newly opened buffers by default?'';
+            default = null;
+          };
+          styling_type = mkOption {
+            type = nullOr (enum ["simple" "none" "colorscheme"]);
+            description = ''What kind of styling should typst-concealer apply to your typst?'';
+            default = null;
+          };
+          ppi = mkOption {
+            type = nullOr int;
+            description = ''What PPI should typst render at. Plugin default is 300, typst's normal default is 144.'';
+            default = null;
+          };
```

Comment:
`null` default will be converted to its Lua equivalent, nil, if set. If you wish to set those options manually (not strictly necessary in the setupOpts format, it's freeform) then you should set appropriate values as `null` values are converted, not filtered.

Also I'd appreciate if you could older `mkOption`paramters as
1. type
2. default
3. (optional) example
4. description

For reference, description does not need `''...''` but can be set as `"..."` unless you are escaping double quotes inside the string literal. using `''''` adds unnecessary whitespace to resulting markdown.